plumpy/process_comms.py

```diff
@@ -571,10 +588,16 @@ async def _launch(
             init_kwargs = {}
 
         proc_class = self._loader.load_object(process_class)
-        proc = proc_class(*init_args, **init_kwargs)
+        proc: Process = proc_class(*init_args, **init_kwargs)
+
+        if proc.pid in self._process_cache and not self._process_cache[proc.pid].has_terminated():
+            raise exceptions.DuplicateProcess(f'Process<{proc.pid}> is already running')
```

Comment:
I don't think we actually use the "launch" part of the communicator but exclusively the "continue" so I don't think this will matter for our use-case, but isn't it a bit weird to be able to hit a duplicate process when launching it? When you launch it, it is the first time you are creating it and so the pid shouldn't already running as it can when continuing an existing process. I can see how there can still be a clash in generated process ids, but I think that the exception type or at the very least the message should be different from the one in `_continue`.

Reply:
yeh as you mention, we don't use it, so this is only for completeness. I would say though, whether you continue or launch, if there are two processes with the same PID they are duplicates, so I disagree that the message should be any difference.
(if you launch twice with the same PID, this is no different to continuing twice)

Reply:
Not quite sure. I agree that the result of having a duplicate PID is the same, however, the origin would be very different and I think that is important to reflect. When you launch a new PID is created and so it should be unique. If that is not the case, then the ID generating algorithm is fundamentally broken, which is completely different from the case in `continue` where can simply have requested to continue the same process twice.

Reply:
just to have the lat word lol

> however, the origin would be very different and I think that is important to reflect

but then you could see the origin in the traceback

> When you launch a new PID is created

this would not be the case if you specifically set the pid in `init_args` or `init_kwargs`